### PR TITLE
ledger-tool: Move ledger output methods to output.rs

### PR DESCRIPTION
**Pull request by steviez**
#### Summary of Changes
We have a dedicated file for CLI output, so move the various output functions there.

There is probably work to be done to clean up these `output_*()` functions; however, I think that should be deferred for the sake of this PR as I have another PR that I would like to have consume this one.

### CODE REVIEW

File: ledger-tool/src/ledger_utils.rs [hunk 1]
```diff
@@ -551,3 +551,13 @@ pub fn open_genesis_config_by(ledger_path: &Path, matches: &ArgMatches<'_>) -> G
...
+
+pub fn get_program_ids(tx: &VersionedTransaction) -> impl Iterator<Item = &Pubkey> + '_ {
+    let message = &tx.message;
+    let account_keys = message.static_account_keys();
+
+    message
+        .instructions()
+        .iter()
+        .map(|ix| ix.program_id(account_keys))
+}
```
Comment (CriesofCarrots) [inline ledger-tool/src/ledger_utils.rs:563]: Aside: seems like this ought to be a method on `VersionedTransaction`

File: ledger-tool/src/output.rs [hunk 2]
```diff
@@ -303,3 +312,238 @@ impl EncodedConfirmedBlockWithEntries {
...
+}
+
+pub fn output_slot(
+    blockstore: &Blockstore,
+    slot: Slot,
+    allow_dead_slots: bool,
+    method: &OutputFormat,
+    verbose_level: u64,
+    all_program_ids: &mut HashMap<Pubkey, u64>,
+) -> std::result::Result<(), String> {
```
Comment (CriesofCarrots) [inline ledger-tool/src/output.rs:406]: Fwiw, I've no objection to changing the import definitions to require `fmt::Result` to be made explicit and allow the std Result to use the shorthand. Seems fair for std to be default.

**Review by CriesofCarrots — DISMISSED** [review]
Lgtm! I don't feel strongly, but if you want to import `std::result::Result` in `output.rs`, I'm happy to re-review.

File: ledger-tool/src/ledger_utils.rs (hunk 1 above)
Comment (steviez) [reply ledger-tool/src/ledger_utils.rs:563]: Good point - I've made a note to look into this as a follow-on task

File: ledger-tool/src/output.rs (hunk 2 above)
Comment (steviez) [reply ledger-tool/src/output.rs]: > Seems fair for std to be default.

Good callout and I agree with this sentiment; I'll make the change now

**Review by steviez — COMMENTED** [review]
I rebased on tip of master to pick up latest, but only change was the extra commit to swap the "default Result"

**Review by CriesofCarrots — APPROVED** [review]

**Pull request merged**